src/rest.js

```diff
@@ -104,8 +84,67 @@ function connect(uid) {
         });
     };
 
+    connection.monitor = function(path, callback, return_raw = false) {
+        return new Promise((resolve, reject) => {
```

Comment:
Just for reference and what I recently stumbled on [Promise.withResolvers()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers). You don't have to switch but it is nice if you want to de-indent code no real benefit here